src/broadcast/macos/macosmediaplayerservice.mm

```diff
@@ -0,0 +1,303 @@
+#include "broadcast/macos/macosmediaplayerservice.h"
+#include "library/autodj/autodjprocessor.h"
+#include "library/coverartcache.h"
+#include "mixer/deck.h"
+#include "mixer/playerinfo.h"
+#include "mixer/playermanager.h"
+#include "track/track.h"
+#include "track/track_decl.h"
+#include "util/assert.h"
+
+#import <AppKit/AppKit.h>
+#import <MediaPlayer/MediaPlayer.h>
+
+#include <QPixmap>
+#include <cstdlib>
+
+constexpr int coverArtSize = 128;
+
+namespace {
+
+MPRemoteCommandHandlerStatus commandHandlerStatusFor(bool success) {
+    return success ? MPRemoteCommandHandlerStatusSuccess
+                   : MPRemoteCommandHandlerStatusCommandFailed;
+}
+
+void setCoverArt(NSMutableDictionary* nowPlayingInfo, const QPixmap& pixmap) {
+    if (@available(macOS 10.13.2, *)) {
+        // Convert the Qt image to a Cocoa image
+        CGImageRef cgImage = pixmap.toImage().toCGImage();
+        NSImage* nsImage = [[NSImage alloc] initWithCGImage:cgImage
+                                                       size:NSZeroSize];
+
+        // Wrap the converted image in an artwork object as required
+        // Since the block escapes, we need to de-stackify it with 'copy'
+        auto fetchArtworkImage = ^NSImage*(CGSize size) {
+            return nsImage;
+        };
+        MPMediaItemArtwork* artwork = [[MPMediaItemArtwork alloc]
+                initWithBoundsSize:CGSizeMake(coverArtSize, coverArtSize)
+                    requestHandler:[fetchArtworkImage copy]];
+        [nowPlayingInfo setObject:artwork forKey:MPMediaItemPropertyArtwork];
+    }
+}
+
+} // anonymous namespace
+
+MacOSMediaPlayerService::MacOSMediaPlayerService(
+        PlayerManagerInterface& pPlayerManager) {
```

Comment:
There are a few design questions that might be interesting though, especially for the "interface PR": Given that this service also handles control in both directions (play/pause, seeking), the question would be if the scrobbling service interface should support that too. Currently the `MacOSMediaPlayerService` simply takes a play manager and then wires itself up to that, but perhaps we'd want to take the opportunity and abstract over that too.